service/matching/task_queue_partition_manager.go

```diff
@@ -85,6 +101,28 @@ func newTaskQueuePartitionManager(
 		cachedPhysicalInfoByBuildId: nil,
 	}
 
+	dynamicRateBurst := quotas.NewMutableRateBurst(
+		defaultTaskDispatchRPS,
+		int(defaultTaskDispatchRPS),
+	)
+	dynamicRateLimiter := quotas.NewDynamicRateLimiter(
+		dynamicRateBurst,
+		defaultTaskDispatchRPSTTL,
+	)
+	limiter := quotas.NewMultiRateLimiter([]quotas.RateLimiter{
+		dynamicRateLimiter,
+		quotas.NewDefaultOutgoingRateLimiter(
+			tqConfig.AdminNamespaceTaskQueueToPartitionDispatchRate,
+		),
+		quotas.NewDefaultOutgoingRateLimiter(
+			tqConfig.AdminNamespaceToPartitionDispatchRate,
+		),
+	})
+
+	pm.dynamicRateBurst = dynamicRateBurst
+	pm.dynamicRateLimiter = dynamicRateLimiter
+	pm.rateLimiter = limiter
```

Comment:
just assign these directly?
```suggestion
	pm.dynamicRateBurst = quotas.NewMutableRateBurst(
		defaultTaskDispatchRPS,
		int(defaultTaskDispatchRPS),
	)
	pm.dynamicRateLimiter = quotas.NewDynamicRateLimiter(
		pm.dynamicRateBurst,
		defaultTaskDispatchRPSTTL,
	)
	pm.rateLimiter = quotas.NewMultiRateLimiter([]quotas.RateLimiter{
		pm.dynamicRateLimiter,
		quotas.NewDefaultOutgoingRateLimiter(
			tqConfig.AdminNamespaceTaskQueueToPartitionDispatchRate,
		),
		quotas.NewDefaultOutgoingRateLimiter(
			tqConfig.AdminNamespaceToPartitionDispatchRate,
		),
	})
```